File: js/kattis/planktonfarm/planktonfarm.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on('line', (i) => {
    rl.on('line', (j) => {
        rl.on('line', (k) => {
            var b = parseInt(i);
            var a = parseInt(j);
            var c = k.split(" ").map(x => parseInt(x)).sort();
            console.log(c.sort());
        });
    });
});

File: rust/kattis/planktonfarm/planktonfarm.rs
fn main() {
    let mut s: String = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    let a: i32 = s.trim().parse().unwrap();
    s = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    let _b: i32 = s.trim().parse().unwrap();
    s = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    let mut c: Vec<i32> = s
        .split(' ')
        .map(|x| x.trim().parse().unwrap())
        .collect::<Vec<i32>>();
    c.sort();
    let mut p1=0;
    let mut p2=c.len() - 1;
    let mut ans=0;
    while p1 < p2 {
        if c[p1] * c[p2] >= a {
            p2 -= 1;
            ans += 1;
        }
        p1 += 1;
    }
    println!("{}", ans);
}
